docusaurus/docs/reactnative/guides/message_input_customization.mdx

```diff
@@ -229,4 +232,164 @@ const additionalTextInputProps = useMemo(() => {
+    <td align='center'>Photo library</td>
+  </tr>
+</table>
+
+Channel component takes a prop [AttachButton](../core-components/channel.mdx#attachbutton),
+which can be used to override the default attach button next to input box as following. This prop can be used to
+override default `onPress` handler for the `AttachButton` component.
+
+```tsx
+import { AttachButton, Channel } from 'stream-chat-react-native';
+
+const CustomAttachButton = () => {
+  const onPressHandler = () => {
+    // Custom handling of onPress action on AttachButton
+  };
+
+  return <AttachButton handleOnPress={onPressHandler} />;
+}
+
+<Channel AttachButton={CustomAttachButton} />
+
+```
+
+Lets implement the `onPressHandler` to open action sheet showing options to choose attachment from Photo Library, Camera or Files.
+We will be using [`@expo/react-native-action-sheet`](https://github.com/expo/react-native-action-sheet)
+package for action sheet implementation in this example.
+
+```sh
+yarn add @expo/react-native-action-sheet
+```
+
+Please read through the documentation of [`@expo/react-native-action-sheet`](https://github.com/expo/react-native-action-sheet#a-basic-actionsheet-setup) for more details.
+
+```tsx {2,5,8-27,33,35}
+import { AttachButton, Channel } from 'stream-chat-react-native';
+import { ActionSheetProvider, useActionSheet } from '@expo/react-native-action-sheet';
+
+const CustomAttachButton = () => {
+  const { showActionSheetWithOptions } = useActionSheet();
+
+  const onPressHandler = () => {
+    // Same interface as https://facebook.github.io/react-native/docs/actionsheetios.html
+    showActionSheetWithOptions(
+      {
+        cancelButtonIndex: 3,
+        destructiveButtonIndex: 3,
+        options: ['Photo Library', 'Camera', 'Files', 'Cancel'],
+      },
+      (buttonIndex) => {
+        switch (buttonIndex) {
+          case 0:
+            break;
+          case 1:
+            break;
+          case 2:
+            break;
+          default:
+            break;
+        }
+      },
+    );
+  };
+
+  return <AttachButton handleOnPress={onPressHandler} />;
+};
+
+<ActionSheetProvider>
+  <Channel AttachButton={CustomAttachButton} />
+</ActionSheetProvider>
+```
+
+Now lets hook the relevant action handlers for buttons within actionsheet.
+[`react-native-image-crop-picker`](https://github.com/ivpusic/react-native-image-crop-picker) provides an image picker functionality
+with good support for configurable compression, multiple images and cropping. And selected images from image picker or camera can be wired in
+with upload previews ([`ImageUploadPreview`](../ui-components/image_upload_preview.mdx)) within `MessageInput` component
+by using a function `uploadNewImage` provided by [`MessageInputContext`](../contexts/message_input_context.mdx#uploadnewimage).
+
+For file attachments, SDK by default uses native file picker. You can reuse that functionality by adding `pickFile` function
```

Comment:
For file attachments, the SDK defaults to the native file picker. That can be reused by adding the... provided by the...